Nouvelle tête pour les messages

AmauryCarrade opened the pull request:
Cette PR prend en compte [les propositions de Sandhose sur le forum](https://zestedesavoir.com/forums/sujet/9274/nouvelle-tete-pour-les-messages) pour améliorer l'affichage des messages sur le site. L'objectif est de les rendre plus compacts, en limitant les actions affichées afin de mettre en évidence la principale, mais aussi (en plus de ce que proposais Sandhose) de clairement séparer les actions personnelles et administratives.

~~Des raffinements sont encore nécessaires pour avoir quelque chose de totalement satisfaisant.~~ C'est bon.

~~Aussi, les tests ne sont pas encore à jour.~~ Tout est vert !


### Aperçus

#### Les messages

Sur nos messages, l'action mise en avant est la modification ; sur ceux des autres, la citation (même si on est staff et qu'on a les droits de les modifier).

![image](https://user-images.githubusercontent.com/1417570/95449676-607bb900-0965-11eb-8637-f8fce6a6f753.png)

#### Le menu des message

Ils s'ouvrent au survol. L'élément surligné est simplement celui survolé.

##### Sans droits, message de quelqu'un d'autre.

![image](https://user-images.githubusercontent.com/1417570/95450434-94a3a980-0966-11eb-9e9e-15fef287d167.png)

##### Sans droits, son message.

![image](https://user-images.githubusercontent.com/1417570/95450380-7fc71600-0966-11eb-92d9-7e1536e9d265.png)

##### Avec droits, message de quelqu'un d'autre.

![image](https://user-images.githubusercontent.com/1417570/95450535-b69d2c00-0966-11eb-9f56-f4ac59024f48.png)

##### Avec droits, son message.

![image](https://user-images.githubusercontent.com/1417570/95450665-e4827080-0966-11eb-88c8-b679fba00075.png)


### Contrôle qualité

_Le travail n'étant pas terminé, il n'est pas pertinent de faire une QA pour le moment._

- Lancez le site avec `make run`.
- Allez sur une page contenant des messages (forum, MP, puis commentaires de contenus).
- Pour chacune, vérifiez que : 
  - l'affichage est correct ;
  - les actions affichées sont les bonnes, il n'en manque pas et il n'y en a pas en trop (par exemple, affichées alors qu'on a pas les droits) ;
  - l'action principale (celle visible sans ouvrir le menu) est logique ;
  - les actions fonctionnent.
- Vérifiez que sur mobile l'affichage est également cohérent.

AmauryCarrade committed:
Nouvelle mise en forme des messages (bureau)

Reste à faire la version mobile

AmauryCarrade committed:
Accessibilité des nouveaux messages

La navigation au clavier est supportée sur le menu déroulant.

AmauryCarrade committed:
“Cette réponse m'a aidé” est réduit pour l'équipe

- Lorsqu'un membre de l'équipe voit un message dans un sujet qu'il n'a
  pas créé, le bouton « Cette réponse m'a aidé » est réduit.

AmauryCarrade committed:
La date est raccourcie sur les petits écrans

- Correction d'un fonctionnement antérieur qui ne marchait plus.

AmauryCarrade committed:
Correction de l'affichage du bouton « m'a aidé »

- Le bouton était déséquilibré lorsqu'affiché sans texte.

AmauryCarrade committed:
Replaced “is author” with a feather icon

- CC BY FontAwesome (TODO add attribution)

AmauryCarrade committed:
Added CSS tooltips and used them for the messages.

- Added CSS tooltips, to use when we need a directly displayed tooltip,
  but don't want to use the JS ones.
- Used these to display “Subject author” and the post edition date,
  on hover, without delay.
- These would be usable for other things as well.

AmauryCarrade committed:
Inclusivité de l'interface des messages

AmauryCarrade committed:
Les infobulles CSS marchent aussi avec data-tooltip

- Si c'est pertinent, on peut utiliser aria-label, mais dans le cas
  contraire, par exemple si l'infobulle est annexe et non descriptive
  de l'ensemble de l'élément, data-toolip peut être utilisé.

AmauryCarrade committed:
Attribution de FA

AmauryCarrade committed:
Adaptation mobile des messages

- Les messages s'affichent correctement sur un téléphone portable ou
  autre petit écran.
- La date et l'info de modification sont déplacées sous le message,
  et la date de modification est dans une modale.
- Sur les très petits écrans, le bouton de citation/modification est
  réduit, afin de laisser la place aux casquettes, si besoin.

Technique

- Ajout d'une media-query `tiny` (et d'une mixin qui va avec) pour les
  très petits écrans.
- Réorganisation du CSS des messages afin que les adaptations mobiles
  soient dans les styles directement, et non à la fin du fichier.

AmauryCarrade commented:
Et voilà la version mobile.

![Version mobile des messages](https://user-images.githubusercontent.com/1417570/103417703-0d905700-4b8c-11eb-8b2e-88dfe479239e.png)

Ici, il reste ceci à faire.

- [x] Corriger les tests qui ne passent plus (c'est probablement pas grand chose, à première vue).
- [x] Documenter la mixin `horizontal-list`.
- [x] Documenter la mixin `tiny`.
- [x] Documenter les infobulles CSS.

J'aimerais également discuter ces points.

- Afin d'intégrer des icônes SVG _directement_ dans le HTML (ce qui permet de les styliser en CSS notamment, et qui évite une requête HTTP), j'ai, un peu naïvement, ajouté un dossier de gabaris pour pouvoir les `{% include %}`, en l'occurrence le dossier `assets/images/inline-svg`. Est-ce que cette façon de faire est validée ? À défaut, comment feriez-vous ? C'est un besoin que l'on aura de toute façon si on veut passer toutes les icônes en SVG.
- J'ai ajouté une seconde façon d'avoir des infobulles “style ZdS”, en CSS pur, dans le cas où une ligne de texte non-formatée suffit. Pas d'opposition ? Ça permet de rendre l'icône plume de l'auteur plus compréhensible, en ayant directement l'info. Je m'en suis également servi pour la date de modification d'un message, car c'est une info qu'on peut souvent vouloir connaître, et qu'elle n'est plus directement visible (pour alléger l'interface notamment, et par manque de place un peu). Aussi, ça pourra servir dans d'autres parties du site.

AmauryCarrade commented:
Ce qui reste étant essentiellement de la doc et des tests, c'est déjà regardable, si vous voulez faire des retours pu vérifier que tout est bon.

AmauryCarrade committed:
Amélioration des messages masqués

- Retrait du JS utilisé pour afficher/masquer les messages. À la place,
  l'élément standard `details` est utilisé.
- Amélioration des styles, marges, etc.
- Correction d'un bug de marges si le dernier message d'un vieux sujet
  est masqué.

AmauryCarrade committed:
Amélioration des messages masqués mobile

- Correction de marges
- Correction d'autres marges
- Retrait de marges
- Ajout de marges
- Suppression de bouts d'avatars récalcitrants
- J'ai aussi ajouté une marge

AmauryCarrade committed:
Ajout de la documentation de `horizontal-list`.

AmauryCarrade committed:
Ajout de la documentation de la largeur/mixin `tiny`.

AmauryCarrade committed:
Modification des infobulles CSS et documentation.

- Par défaut, les infobulles CSS utilisent l'attribut `aria-describedby`.
- Il est toujours possible d'utiliser `aria-label` ou `data-tooltip`.
- Ajout de la documentation de ces infobulles.

AmauryCarrade committed:
Correction des tests

- Le test zds.forum.tests.tests_views.MessageActionTest.test_hide ne
  testait plus le bon élément suite aux changements de la façon
  d'afficher le message masqué.
- Le test zds.mp.tests.tests_views.TopicViewTest.test_available_actions
  révélait effectivement un oubli : il était possible demander à masquer
  un message dans un MP.

AmauryCarrade commented:
Cette PR est prête.

AmauryCarrade committed:
Le masquage était bien trop masqué

- Certains messages masquables n'avaient pas l'option de disponible.

AmauryCarrade committed:
Ajout du karma sur la version mobile

- Le karma global de l'utilisateur est affiché dans le pied du message,
  sur mobile.

AmauryCarrade committed:
Renommage d'une classe incohérente

- `date-mobile` s'appelle désormais `metadata-mobile`, car il y a bien
  plus que juste la date dans cet élément désormais.

Situphen reviewed (changes requested):
**QA NOK** Voici les petites coquilles que j'ai trouvé : 

- [x] Lorsque la signature est très longue, cliquer dessus ne permet plus de la voir en entier (que ce soit sur ordinateur ou sur mobile)
- [x] Pour marquer un message comme spam potentiel il y a deux boutons : un à côté des pouces et un dans le menu déroulant. De mon point de vue, il ne devrait y avoir que celui du menu déroulant étant donné le peu de fois où on marque un message comme spam potentiel. Si jamais on décide de garder les deux, alors il faut unifier leur comportement car actuellement appuyer sur un bouton ne modifie pas le texte de l'autre.
- [x] ~~Par rapport à la version en production, le texte du bouton « Cette réponse m'a aidé » n’apparaît plus sur ordinateur. Je ne pense pas que ce soit judicieux. Si le texte prend trop de place, alors on peut le mettre dans le menu déroulant.~~ Déplacer “Réponse utile” dans le menu déroulant si on est modérateur et que ce n'est pas notre sujet.
- [x] Connecté avec un compte modérateur, il n'y a pas le bouton Citer, alors que celui-ci est bien présent en tant que simple utilisateur
- [x] Sur mobile, cliquer sur "Modifié" ou "Modifié par X" affiche une fenêtre modale. Je pense qu'il serait mieux d'avoir simplement une info-bulle comme pour « Auteur ou autrice du sujet » en tant que simple utilisateur (ni auteur du message ni staff) et directement le lien vers l'historique de modification pour l'auteur du message ou le staff. En fait, garder le même comportement que sur ordinateur.

**Je suis extrêmement fan de cette nouvelle version !** On pourrait presque enlever la fonctionnalité « Un ou plusieurs messages ont été masqués » tellement les messages masqués ne prennent pas beaucoup de place et se fondent bien dans la masse.

AmauryCarrade commented:
Merci pour la QA ! Je reviens sur quelques points ; pour tout le reste, c'est effectivement quelque chose à corriger et je n'ai rien à ajouter.

> Pour marquer un message comme spam potentiel il y a deux boutons : un à côté des pouces et un dans le menu déroulant. De mon point de vue, il ne devrait y avoir que celui du menu déroulant étant donné le peu de fois où on marque un message comme spam potentiel. Si jamais on décide de garder les deux, alors il faut unifier leur comportement car actuellement appuyer sur un bouton ne modifie pas le texte de l'autre.

Ce n'est effectivement pas normal. Je n'avais pas constaté ça lors de la fusion, je vais vérifier… Mais c'est simplement un souci de fusion faite un peu trop vite (il faut dire qu'elle était particulièrement _tricky_).

> Par rapport à la version en production, le texte du bouton « Cette réponse m'a aidé » n’apparaît plus sur ordinateur. Je ne pense pas que ce soit judicieux. Si le texte prend trop de place, alors on peut le mettre dans le menu déroulant.

Tu as dû mal voir. Ou plus précisément, ne voir que partiellement.

J'ai en effet changé ce comportement, mais de la façon suivante : 
- sur ses propres sujets, on voit le bouton avec le libellé ;
- sur les sujets des autres, si on a les droits, on voit le bouton réduit.

Ça permet de limiter le bruit et de marquer une différence visuelle entre les cas où on agit pour nous, et ceux où on agit pour autrui.

Par contre si le bouton est systématiquement réduit, alors oui y'a un souci. Quand j'y pense, ajouter un test de ce comportement ferait sens.

> Sur mobile, cliquer sur "Modifié" ou "Modifié par X" affiche une fenêtre modale. Je pense qu'il serait mieux d'avoir simplement une info-bulle comme pour « Auteur ou autrice du sujet » en tant que simple utilisateur (ni auteur du message ni staff) et directement le lien vers l'historique de modification pour l'auteur du message ou le staff. En fait, garder le même comportement que sur ordinateur.

Dans ce cas, en tant que modérateur ou auteur, on ne pourrait plus facilement voir l'infobulle. Enfin, il faut maintenir appuyé au lieu de tapoter, ce qui n'est pas des plus intruitifs. Est-ce souhaitable ?

L'usage d'une modale évitait précisément ce comportement.

> Je suis extrêmement fan de cette nouvelle version !

Merci <3 Soulignons qu'une partie du mérite revient à @sandhose, qu'a posé des bases de ce nouveau design :) .

Situphen commented:
> > Par rapport à la version en production, le texte du bouton « Cette réponse m'a aidé » n’apparaît plus sur ordinateur. Je ne pense pas que ce soit judicieux. Si le texte prend trop de place, alors on peut le mettre dans le menu déroulant.
> 
> Tu as dû mal voir. Ou plus précisément, ne voir que partiellement.
> 
> J'ai en effet changé ce comportement, mais de la façon suivante :
> 
>     * sur ses propres sujets, on voit le bouton avec le libellé ;
> 
>     * sur les sujets des autres, si on a les droits, on voit le bouton réduit.
> 
> 
> Ça permet de limiter le bruit et de marquer une différence visuelle entre les cas où on agit pour nous, et ceux où on agit pour autrui.
> 
> Par contre si le bouton est systématiquement réduit, alors oui y'a un souci. Quand j'y pense, ajouter un test de ce comportement ferait sens.

OK alors en effet le comportement est bien celui que tu indiques. Je me demande si on ne devrait pas créer un bouton « Marquer comme utile » dans la partie Modération du menu déroulant au lieu du bouton réduit pour ceux qui ont les droits.

> > Sur mobile, cliquer sur "Modifié" ou "Modifié par X" affiche une fenêtre modale. Je pense qu'il serait mieux d'avoir simplement une info-bulle comme pour « Auteur ou autrice du sujet » en tant que simple utilisateur (ni auteur du message ni staff) et directement le lien vers l'historique de modification pour l'auteur du message ou le staff. En fait, garder le même comportement que sur ordinateur.
> 
> Dans ce cas, en tant que modérateur ou auteur, on ne pourrait plus facilement voir l'infobulle. Enfin, il faut maintenir appuyé au lieu de tapoter, ce qui n'est pas des plus intruitifs. Est-ce souhaitable ?
> 
> L'usage d'une modale évitait précisément ce comportement.

Ce n'est à mon avis pas gênant si les modérateurs ne peuvent plus voir l'info-bulle car les informations qui sont dedans (auteur, date et heure de la dernière édition) sont aussi présents sur la page d'historique des éditions. En fait la question c'est est-ce qu'il est préférable d'avoir un seul clic et un chargement de page ou deux clics sans chargement de page pour accéder aux informations de l'info-bulle. On peut très bien laisser tel quel et on verra si le Staff veut changer plus tard.

AmauryCarrade commented:
>  Connecté avec un compte modérateur, il n'y a pas le bouton Citer, alors que celui-ci est bien présent en tant que simple utilisateur

Je ne reproduis pas, que ce soit connecté comme `admin` (ici) ou comme `staff`.

Le bouton « Citer » est même celui directement mis en avant sur les messages (sauf les siens) ^^ .

![image](https://user-images.githubusercontent.com/1417570/107582847-5eb75000-6bfa-11eb-8dab-9618039b2c5f.png)

Par contre il y a effectivement des soucis de permissions quand on est `staff` et pas `admin`.

AmauryCarrade commented:
> OK alors en effet le comportement est bien celui que tu indiques. Je me demande si on ne devrait pas créer un bouton « Marquer comme utile » dans la partie Modération du menu déroulant au lieu du bouton réduit pour ceux qui ont les droits.

Je pense que c'est une bonne idée.

> Ce n'est à mon avis pas gênant si les modérateurs ne peuvent plus voir l'info-bulle car les informations qui sont dedans (auteur, date et heure de la dernière édition) sont aussi présents sur la page d'historique des éditions. En fait la question c'est est-ce qu'il est préférable d'avoir un seul clic et un chargement de page ou deux clics sans chargement de page pour accéder aux informations de l'info-bulle. On peut très bien laisser tel quel et on verra si le Staff veut changer plus tard.

Par « laisser comme ça », tu entends comme c'est actuellement dans la PR, ou avec infobulle pour les personnes n'ayant pas le droit de voir l'historique, et modale pour celles ayant ce droit ?

Situphen commented:
> Par « laisser comme ça », tu entends comme c'est actuellement dans la PR, ou avec infobulle pour les personnes n'ayant pas le droit de voir l'historique, et modale pour celles ayant ce droit ?

Je dirais « infobulle pour les personnes n'ayant pas le droit de voir l'historique, et modale pour celles ayant ce droit » car une info-bulle c'est quand même mieux quand c'est possible :)

> > Connecté avec un compte modérateur, il n'y a pas le bouton Citer, alors que celui-ci est bien présent en tant que simple utilisateur
> 
> Je ne reproduis pas, que ce soit connecté comme `admin` (ici) ou comme `staff`.
> 
> Le bouton « Citer » est même celui directement mis en avant sur les messages (sauf les siens) ^^ .

Je viens de comprendre : je n'avais pas vu que le sujet en question était verrouillé ! Donc effectivement il n'y a pas de soucis : bouton Citer sauf si sujet verrouillé ou si on vient de poster.

AmauryCarrade committed:
Inutile d'afficher le bouton d'antispam quinze fois.

AmauryCarrade committed:
Les signatures s'ouvrent à nouveau au clic.

- Ouverture au clic des signatures qui dépassent une
  ligne.
- Ce système n'utilise plus de javascript.

AmauryCarrade committed:
“Réponse utile” dans le menu si on est modérateur⋅trice.

- Lorsqu'un modérateur ou une modératrice consulte un
  sujet dont iel n'est pas l'auteur, le bouton “cette
  réponse est utile” est déplacé dans le menu de dépas-
  sement. Lorsqu'iel consulte un message dont iel est
  l'auteur⋅trice, par contre, le bouton est toujours
  affiché au même endroit.

AmauryCarrade committed:
Simple infobulle sur la date sur mobile sans droits

- La date de modification est affichée uniquement avec
  une infobulle si on n'a pas les droits d'afficher
  l'historique.

AmauryCarrade commented:
Les points adressés ont été corrigés.

Nouvelle QA : vérifier que le bouton pour marquer un message comme utile s'affiche au bon endroit quand on est staff (sous le message sur nos sujets et dans le menu sinon), et toujours sous le message si on n'est pas staff (pas dans le menu).

Situphen reviewed (changes requested):
**QA NOK**

- [x] (Important) Entre la version ordinateur et la version tablette, la case à cocher de la signature apparait
  ![image](https://user-images.githubusercontent.com/6664636/107860586-489ad100-6e40-11eb-9dca-ae4dbceb6a56.png)
- [x] (Peu important) Entre la version ordinateur et la version tablette, la marge à gauche du `<summary>` est inexistante
  ![image](https://user-images.githubusercontent.com/6664636/107860692-fc03c580-6e40-11eb-8f87-d267af8b12a6.png)
- [x] (Moyennement important) Sur mobile en tant que simple utilisateur, il n'est pas précisé quand l'édition d'un message a été fait pour un modérateur (il faudrait que ça apparaisse dans l'info-bulle)
- [x] (Peu important) Impossible de lire en entier la signature si l'on n'est pas connecté
- [x] (Important) Sur mobile, quand on est l'auteur du sujet, le bouton « Cette réponse m'a été utile » prend trop de place. On pourrait raccourcir le nom en « Réponse utile », ou alors remettre la date à côté du pseudo (mais là encore ça peut bloquer avec des pseudos longs :/)

Situphen commented on assets/scss/components/_topic-message.scss:
```diff
@@ -331,8 +600,35 @@ div.msg-are-hidden {
+                // These styles are only applied to summaries as they
+                // are used to make this visibly clickable. Theses should
+                // not be applied to p.message-hidden-reason, displayed to
+                // users without the right to see the hidden message.
```
```suggestion
                // These styles are only applied to summaries as they
                // are used to make this visibly clickable. These should
                // not be applied to p.message-hidden-reason, displayed to
                // users without the right to see the hidden message.
```

Situphen commented on assets/scss/components/_topic-message.scss:
```diff
@@ -615,6 +950,17 @@ form.topic-message {
 }
 
+
+@include desktop {
+    .topic-message {
+
+        .message {
+
+        }
+    }
+}
```
Oubli ?

AmauryCarrade replied:
Yup

Situphen commented on assets/scss/components/_topic-message.scss:
```diff
@@ -615,6 +950,17 @@ form.topic-message {
+    }
+}
+
+/*
```
(Tout le commentaire) Oubli ?

AmauryCarrade replied:
Yup

J'étais convaincu de l'avoir supprimé d'ailleurs

Situphen commented on doc/source/front-end/elements-specifiques-au-site.rst:
```diff
@@ -128,6 +128,54 @@ Il est possible de créer des modales en Javascript. Exemple:
+À ne pas confondre avec `les infobulles en JavaScript <helpers-js.html#tooltip>`_, les infobulles CSS ont la même apparence
+mais sont beaucoup plus légères et simples à utiliser. Par contre, elles ne peuvent contenir de contenu HTML riche, ni ne
+réagir au clic. Elles sont donc efficace pour des infobulles purement informatives, mais qui sont suffisamment importantes
+pour être affichées directement au survol, et non en attendant une seconde ou deux (comme un attribut ``title``).
```
```suggestion
À ne pas confondre avec `les infobulles en JavaScript <helpers-js.html#tooltip>`_, les infobulles CSS ont la même apparence
mais sont beaucoup plus légères et simples à utiliser. Par contre, elles ne peuvent contenir de contenu HTML riche, ni ne
réagir au clic. Elles sont donc efficaces pour des infobulles purement informatives, mais qui sont suffisamment importantes
pour être affichées directement au survol, et non en attendant une seconde ou deux (comme un attribut ``title``).
```

Situphen commented on doc/source/front-end/helpers-js.rst:
```diff
@@ -18,6 +18,11 @@ La tooltip sera par défaut au dessus de l'élément, mais ira automatiquement e
+.. note::
+
+   Si vous n'avez besoin que d'infobulles simples, sans avoir à y mettre du contenu HTML riche ou à réagir au clic ou
+   autre sur ces dernière, `préférez utiliser les plus simples et légères infobulles CSS, d'apparence identique <elements-specifiques-au-site.html#les-infobulles-css>`_.
```
```suggestion
   Si vous n'avez besoin que d'infobulles simples, sans avoir à y mettre du contenu HTML riche ou à réagir au clic ou
   autre sur ces dernières, `préférez utiliser les plus simples et légères infobulles CSS, d'apparence identique <elements-specifiques-au-site.html#les-infobulles-css>`_.
```

Situphen commented on doc/source/front-end/helpers-scss.rst:
```diff
@@ -127,3 +129,43 @@ mêmes non plus.
+Listes au format horizontal (``horizontal-list``)
+=================================================
+
+Cette mixin CSS sert à transformer n'importe quelle liste (numérotée ou non, de toutes façon les puces seront masquées)
+en liste horizontale.
```
```suggestion
Cette mixin CSS sert à transformer n'importe quelle liste (numérotée ou non, de toutes façons les puces seront masquées)
en liste horizontale.
```

AmauryCarrade replied:
Plutôt : 

```suggestion
Cette mixin CSS sert à transformer n'importe quelle liste (numérotée ou non, de toute façon les puces seront masquées)
en liste horizontale.
```

Situphen commented on templates/misc/message.part.html:
```diff
@@ -53,198 +57,330 @@
+                {% spaceless %}
+                    <aside class="message-actions">
+                        <ul tabindex="0">
+                            {# User actions on his own message #}
```
```suggestion
                            {# User actions on their own message #}
```

Situphen commented on templates/misc/message.part.html:
```diff
@@ -53,198 +57,330 @@
+                                {% endif %}
+                            {% endif %}
+
+                            {# Actions related to someother message (non-administrative) #}
```
```suggestion
                            {# Actions related to some other message (non-administrative) #}
```

Situphen reviewed (changes requested):
Je viens de réaliser que tu as modifié « Cette réponse a aidé l'auteur » en « Cette réponse a été utile ». Effectivement cette forme non genrée est plus inclusive mais je suis tombé sur [ce message de entwanne](https://beta.zestedesavoir.com/forums/sujet/9274/nouvelle-tete-pour-les-messages/#p162023) que je trouve pertinent (et que tu as d'ailleurs plussoyé :P) : 

> Je ne pense pas que « Cette réponse m’a aidé » puisse être remplacé par « Réponse utile ». « Ce message est utile » existe déjà, c’est le bouton +1.
> 
> Qu’apporte « Cette réponse m’a aidé » par rapport au +1 ? La mise en valeur du message, par laquelle l’auteur du sujet reconnaît la pertinence de la réponse.
> 
> Je pense que c’est à ça qu’il faut réfléchir pour trouver un nouveau nom, peut-être quelque chose comme « Valoriser ».

La réponse de cepus est elle aussi pertinente (et a reçu plus de pouces verts) : 

> "Cette réponse m’a aidé" est uniquement disponible pour le PO. Dans un contexte de forums d’entraides où des gens viennent notamment depuis des moteurs de recherche sur des questions qu’ils se posent eux aussi, je trouve que "Réponse utile" a plein de sens.
> 
> Le +1 reste à disposition des lecteurs connectés, bien sûr.
> 
> (J’insiste pas énormément, je trouve "Réponse utile" mieux que "Cette réponse m’a aidé" mais je me battrai pas pour changer ça !)

Je n'ai pas d'avis définitif là dessus mais je pense que ça mérite réflexion (et d'autres avis extérieurs ?).

AmauryCarrade committed:
Ajout d'un test sur la position de “Réponse utile”

- La position (et la présence) du bouton “Cette réponse
  est utile” est vérifiée grâce à un test.

AmauryCarrade committed:
Pas de case à cocher dans les signatures

- Les signatures s'affichent correctement même sur
  tablette.
- Retrait d'un CSS totalement inutile (lié).

AmauryCarrade committed:
Marges sur tablette du contenu d'un message masqué

AmauryCarrade committed:
Abolition des privilèges des signatures

- On peut afficher les signatures au complet même sans
  être connecté⋅e.

AmauryCarrade committed:
Affichage réduit de “réponse utile” sur petits écrans

AmauryCarrade committed:
Amélioration de l'infobulle d'édition sur mobile

- En tant que simple utilisateur, l'infobulle affiche
  l'éditeur d'un message, lorsqu'il est différent de son
  auteur.

AmauryCarrade commented on assets/scss/main.scss:
```diff
@@ -5,13 +5,14 @@
 ------------------------*/
 @import "normalize.css/normalize";
-@import "highlight.js/styles/idea";
+// @import "highlight.js/styles/idea";
```
Précision : modif temporaire liée à mon installation et au test de #6010, j'avais prévu de ne pas le commit mais j'ai oublié de l'exclure. Sera annulé dés que j'aurai une meilleure connexion internet permettant de rafraîchir les paquets npm chez moi.

AmauryCarrade committed:
Apply suggestions from code review

Co-authored-by: Situphen <Situphen@users.noreply.github.com>

AmauryCarrade committed:
Code mort

AmauryCarrade commented:
> Je viens de réaliser que tu as modifié « Cette réponse a aidé l'auteur » en « Cette réponse a été utile ». Effectivement cette forme non genrée est plus inclusive mais je suis tombé sur ce message de entwanne que je trouve pertinent (et que tu as d'ailleurs plussoyé :P) :

@Situphen Effectivement je l'avais plussoyée, mais après réflexion, je trouve celle de vhf bien plus pertinente, et j'étais donc de cet avis. Comme elle avait en plus plus de pouces verts, je partais là dessus.

> Je n'ai pas d'avis définitif là dessus mais je pense que ça mérite réflexion (et d'autres avis extérieurs ?).

On peut effectivement solliciter des avis extérieurs !

AmauryCarrade commented:
> On peut effectivement solliciter des avis extérieurs !

Les avis extérieurs semblent dire “c'est très bien comme ça”

AmauryCarrade commented:
Reste un souci de tap à régler quand on est sur mobile : tendance à sélectionner par erreur le second élément si on tape trop bas.

AmauryCarrade committed:
Retour de la coloration syntaxique (oups)

AmauryCarrade committed:
Amélioration du masquage des messages

- Le nombre de messages masqués est affiché, lorsque réduit.

AmauryCarrade committed:
Fixed touch interactions with messages menu (I guess)

- On touch devices, the opening and closing of the menu is handled
  manually; we no longer rely on the “hover-to-touch-event” automatic
  conversion.

AmauryCarrade committed:
Renamed message-accessibility.js for coherence.

AmauryCarrade committed:
Fixed JS error. This being said, the quote autoscroll no longer work.

AmauryCarrade commented:
Normalement tout a été corrigé (avec même un bonus) mais j'attends de pouvoir tester en bêta sur un vrai téléphone pour en être sûr.

AmauryCarrade commented:
~~Souci de lien du karma d'un membre sur mobile~~ Corrigé

AmauryCarrade committed:
Le lien de karma d'un membre est correct sur mobile.

- Le lien pointait vers la page courante, et non vers la page de
  profil.

AmauryCarrade committed:
Le bouton “réponse utile” s'inverse au clic dans le menu.

- L'inversion du bouton ne fonctionnait que sous le message, et pas
  dans le menu de modération.

AmauryCarrade committed:
Try to fix double-tap to close menu

Situphen reviewed (approved):
**QA OK** Pour moi c'est bon ! On va forcément trouver quelques petites coquilles plus tard mais on les corrigera dans une autre PR si c'est le cas.

Situphen committed:
Merge branch 'dev' into refactor-messages
